### ISSUE TITLE
Test failure: RunIndividualTestCase

### ISSUE THREAD

**Issue by aspnet-hello** [aspnet/KestrelHttpServer#2857]
This test [fails](http://aspnetci/viewLog.html?buildId=539998&buildTypeId=Releases_22xPublic_UbuntuUniverse) occasionally with the following error:
```
System.IO.FileNotFoundException : Could not find file '/mnt/temp/buildTmp/9aa38076-038c-415b-b0e6-5abf3f4062cf.xml'.
   at Interop.ThrowExceptionForIoErrno(ErrorInfo errorInfo, String path, Boolean isDirectory, Func`2 errorRewriter)
   at Microsoft.Win32.SafeHandles.SafeFileHandle.Open(String path, OpenFlags flags, Int32 mode)
   at System.IO.FileStream..ctor(String path, FileMode mode, FileAccess access, FileShare share, Int32 bufferSize, FileOptions options)
   at System.IO.StreamReader..ctor(String path, Encoding encoding, Boolean detectEncodingFromByteOrderMarks, Int32 bufferSize)
   at System.IO.File.InternalReadAllText(String path, Encoding encoding)
   at System.IO.File.ReadAllText(String path)
   at Microsoft.AspNetCore.Server.Kestrel.FunctionalTests.H2SpecCommands.RunTest(String testId, Int32 port, Boolean https, ILogger logger) in /_/test/shared/TransportTestHelpers/H2SpecCommands.cs:line 172
   at Microsoft.AspNetCore.Server.Kestrel.FunctionalTests.Http2.H2SpecTests.RunIndividualTestCase(H2SpecTestCase testCase) in /_/test/Kestrel.Transport.FunctionalTests/Http2/H2SpecTests.cs:line 49
--- End of stack trace from previous location where exception was thrown ---
------- Stdout: -------
| [0.020s] TestLifetime Information: Starting test RunIndividualTestCase-http2/8.1.2.3/3, HTTPS:True, Sends a HEADERS frame that omits ":scheme" pseudo-header field at 2018-08-27T19:21:54
| [0.226s] Microsoft.AspNetCore.Hosting.Internal.WebHost Debug: Hosting starting
| [0.561s] Microsoft.AspNetCore.Hosting.Internal.WebHost Debug: Hosting started
| [0.561s] Microsoft.AspNetCore.Hosting.Internal.WebHost Debug: Loaded hosting startup assembly Libuv.FunctionalTests
| [0.692s] Microsoft.AspNetCore.Server.Kestrel Debug: Connection id "0HLGCA9CPI5UJ" started.
| [1.362s] Microsoft.AspNetCore.Server.Kestrel.Transport.Libuv Debug: Connection id "0HLGCA9CPI5UJ" sending FIN.
| [1.373s] HttpsConnectionAdapter Debug: Authentication of the HTTPS connection timed out.
| [1.376s] Microsoft.AspNetCore.Server.Kestrel.FunctionalTests.Http2.H2SpecTests Debug: Hypertext Transfer Protocol Version 2 (HTTP/2)
|                                                                                         8. HTTP Message Exchanges
|                                                                                           8.1. HTTP Request/Response Exchange
|                                                                                             8.1.2. HTTP Header Fields
|                                                                                               8.1.2.3. Request Pseudo-Header Fields
|                                                                                                   3: Sends a HEADERS frame that omits ":scheme" pseudo-header field
          × 3: Sends a HEADERS frame that omits ":scheme" pseudo-header field
|                                                                                       Error: EOF
| [1.377s] Microsoft.AspNetCore.Hosting.Internal.WebHost Debug: Hosting shutdown
| [1.378s] Microsoft.AspNetCore.Server.Kestrel Debug: Connection id "0HLGCA9CPI5UJ" stopped.
| [1.383s] TestLifetime Information: Finished test RunIndividualTestCase-http2/8.1.2.3/3, HTTPS:True, Sends a HEADERS frame that omits ":scheme" pseudo-header field in 1.3770277s
```
Other tests within that build may have failed with a similar message, but they are not listed here. Check the link above for more info.

This test failed on 2.2.

CC @muratg

This issue was made automatically. If there is a problem contact ryanbrandenburg.

**Comment by aspnet-hello** [comment]
Please use this workflow to address this flaky test issue, including checking applicable checkboxes and filling in the applicable "TODO" entries:

* Is this actually a flaky test?
  * No, this is a regular test failure, fix the test/product (TODO: Link to commit/PR)
  * Yes, proceed below...

* Is this test failure caused by product code flakiness? (Either this product, or another product this test depends on.)
  * [ ] File a bug against the product (TODO: Link to other bug)
  * Is it possible to change the test to avoid the flakiness?
    * Yes? Go to the "Change the test!" section.
    * No?
      * [ ] Disable the test (TODO: Link to PR/commit)
      * [ ] Wait for other bug to be resolved
      * [ ] Wait for us to get build that has the fix
      * [ ] Re-enable our test (TODO: Link to PR/commit)
      * [ ] Close this bug

* Is it that the test itself is flaky? This includes external transient problems (e.g. remote server problems, file system race condition, etc.)
  * Is there is a way to change our test to avoid this flakiness?
    * Yes? Change the test!
      * [ ] Change the test to avoid the flakiness, for example by using a different test strategy, or by adding retries w/ timeouts (TODO: Link to PR/commit)
      * [ ] Run the test 100 times locally as a sanity check.
      * [ ] Close this bug
    * No?
      * Is there any logging or extra information that we could add to make this more diagnosable when it happens again?
        * Yes?
            * [ ] Add the logging (TODO: Link to PR/commit)
        * No?
            * [ ] Delete the test because flaky tests are not useful (TODO: Link to PR/commit)

This comment was made automatically. If there is a problem contact ryanbrandenburg.

**Comment by muratg** [comment]
cc @Tratcher @halter73

**Comment by aspnet-hello** [comment]
RunIndividualTestCase [failed with about the same error](http://aspnetci/viewLog.html?buildId=540065&buildTypeId=Releases_22xPublic_UbuntuUniverse) on 2.2.

This comment was made automatically. If there is a problem contact ryanbrandenburg.

**Comment by halter73** [comment]
@Tratcher Do you mind taking this one?

**Comment by Tratcher** [comment]
3 failures to date, all on the 27th, all on Ubuntu.

Logs:
[12:21:55] :	 [Step 3/3]   KestrelHttpServer     |    | [0.692s] Microsoft.AspNetCore.Server.Kestrel Debug: Connection id "0HLGCA9CPI5UJ" started.
[12:21:55] :	 [Step 3/3]   KestrelHttpServer     |    | [1.362s] Microsoft.AspNetCore.Server.Kestrel.Transport.Libuv Debug: Connection id "0HLGCA9CPI5UJ" sending FIN.
[12:21:55] :	 [Step 3/3]   KestrelHttpServer     |    | [1.373s] HttpsConnectionAdapter Debug: Authentication of the HTTPS connection timed out.

This looks like another manifestation of https://github.com/aspnet/ServerTests/issues/141#issuecomment-417432390, https://github.com/aspnet/KestrelHttpServer/issues/2850, fixed by aspnet/KestrelHttpServer#2851. @halter73 that 1 second timeout could also affect Ssl handshakes, right?

**Comment by halter73** [comment]
Yep. The heartbeat regression could also affect TLS handshake timeouts.
